components/support/base/src/main/java/mozilla/components/support/base/feature/FeatureManager.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.components.support.base.feature
+
+import android.arch.lifecycle.Lifecycle
+import android.arch.lifecycle.LifecycleObserver
+import android.arch.lifecycle.OnLifecycleEvent
+
+class FeatureManager(private val featureList: MutableList<Feature> = mutableListOf()) : LifecycleObserver {
+
+    fun register(feature: Feature) {
```

Comment:
Idea 💡: We could overload the `plusAssign` and `minusAssign` for a more idiomatic way to adding or removing features:
 
```kotlin
class FeatureManager {

        operator fun minusAssign(feature: Feature) {
            register(feature)
        }

        operator fun plusAssign(feature: Feature) {
            unregister(feature)
        }
    }

    fun main() {
        val manager = FeatureManager()
        manager += feature1
        manager -= feature1
    }
```

Reply:
I think the varargs solution should be good enough. I can see us mostly using the manager for (un)registering all features.

Reply:
👍